### DISCUSSION TITLE
Condividere sensori con altra istanza HA

### DISCUSSION THREAD

**Discussion by sixdas** [indomus/forum#2769] [ZigBee, Z-Wave, Thread, Matter]
Ciao a tutti, ho necessità di condividere alcuni sensori zigbee della mia configurazione con un altra. Ecco come è strutturata:

1. Home assistant Container + eclipse-mosquitto + zigbee2mqtt  - ip: 192.168.10.10  (HA 1)
2. HA OS + eclipse-mosquitto + zigbee2mqtt  - ip: 192.168.10.20  (HA 2)

Io pensavo che in HA nell'integrazione mqtt si potevano aggiungere più server in realtà ho scoperto che se ne puo aggiungere solo uno. Quindi cercando un po in rete ho trovato che bisogna creare un bridge tra i due broker e cosi ho fatto, nella configurazione del primo ho messo:

> connection broker2
> address 192.168.10.20:1883
> bridge_attempt_unsubscribe true
> cleansession true
> remote_username mqtt
> remote_password mqtt
> remote_clientid bridgeClientId
> topic # both 0 “”
> 

Funziona, ora mi ritrovo tutti i dispositivi di HA 1 su HA 2 e viceversa.  Ora io vorrei fare in modo che ognuno vede i propri dispositivi e in aggiunta  quelli condivisi.

Es. 

HA 1 = Luce 1 - **Temperature Esterna** - **Luce Cortile** - Luce Cameretta
HA 2 = Luce Cucina - Luce Bagno -  **Temperature Esterna** - **Luce Cortile** - Luce Salone

Sapete darmi una mano?

**Comment by KC-inDomus** [comment]
ma non facevi prima a usare un solo broker e collegarvi sopra i due zigbee2mqtt?

**Reply by sixdas** [reply]
Ci avevo pensato, ma preferisco che le due piattaforme siano separate, perchè nel caso che una delle due venga meno l'altra deve funzionare,  inoltre con un solo broker non avrei sempre lo stesso problema  del bridge , che entrambi vediamo gli stessi dispositivi?

**Reply by snakuzzo** [reply]
Il problema è che tu vedi il broker MQTT come un qualcosa che fa parte di HA, invece non è così.
Il broker MQTT è un broker MQTT... può fare da broker per questa roba così come per altra...a livello architetturale è un elemento a se stante. Se proprio vuoi fare una cosa precisa, dovresti slegarlo completamente dal resto

**Reply by sixdas** [reply]
Lo so che un broker e una cosa a se stante, io quando dico di rimane separate intendo che se metto un unico broker , uso quello mio sul mio nas e poi voglio aggiornare o fare prove, anche la seconda istanza che usa il mio broker deve star ferma O sbaglio qualcosa?

Usando mqtt explorer io ho questa situazione :

![5d7175f3-bde5-4259-8105-16ba481896da](https://github.com/indomus/forum/assets/88273860/622f2bc9-63b9-476b-81b1-f073c25c3077)


Quindi ho gia la separazione tra i tue topic come vedi,  ma in mqtt vedo l'unione di entrambi.

**Reply by KC-inDomus** [reply]
> Ci avevo pensato, ma preferisco che le due piattaforme siano separate, perchè nel caso che una delle due venga meno l'altra deve funzionare,  inoltre con un solo broker non avrei sempre lo stesso problema  del bridge , che entrambi vediamo gli stessi dispositivi?
> 
> 

a parte che le due istanze di Z2M le suddividi logicamente usando topic diversi, ma comunque non ha proprio senso usare due broker

**Reply by sixdas** [reply]
Ok, perche non ha senso? aiutami a capire sicuramente sto ignorando qualcosa

**Reply by snakuzzo** [reply]
Te l'ho spiegato prima

**Reply by KC-inDomus** [reply]
perché il broker ha un ruolo unico, ne installi uno solo sulla tua rete e fa il suo ruolo... quello di [broker mqtt](https://indomus.it/componenti/message-queue-telemetry-transport-mqtt/).

**Reply by sixdas** [reply]
Ok supponiamo che uso un solo broker, poi come faccio a dire che in HA 1 vedo dei dispositivi e sul HA 2 altri?

**Reply by snakuzzo** [reply]
Grazie all'autodiscovery di HA.
Quando abiliti l'integrazione con HA da Zigbee2mqtt, lo stesso zigbee2mqtt pubblicherà una serie di topic di autodiscovery sui quali HA è in ascolto che permetterà la generazione automatica su Ha dei device e delle entità

**Reply by sixdas** [reply]
Quindi devo agire sul singolo dispositivo da Zigbee2mqtt e farlo pubblicare su topic diversi?  Tipo adesso Zigbee2mqtt pubblica sotto il base_topic : zigbee2mqtt

**Reply by snakuzzo** [reply]
sull'altra istanza userai un altro `base_topic` (eg. `zigbee2mqtt-pippo`)

**Comment by sixdas** [comment]
ok è quindi in questa situazione

![5d7175f3-bde5-4259-8105-16ba481896da](https://github.com/indomus/forum/assets/88273860/c775cf0f-62b9-4fe9-9be9-b6fd1f806f76)


Come poi dico che dal zigbee2mqtt-pippo deve prendere solo 2 sensori e non tutto?

**Reply by snakuzzo** [reply]
questo poi lo decidi lato HA...altrimenti dovresti non usare più l'autodiscovery e creare, per tutti i device esposti da zigbee2mqtt , TUTTE le entità MQTT a manella

**Reply by sixdas** [reply]
In ha devo poi specificare quale topic leggere?  se si sai indicarmi come fare oppure qualche guida?

In Zigbee2mqtt  è possibile per il singolo dispositivo cambiare topic?

**Reply by snakuzzo** [reply]
Ogni device pubblica dei topic univoci formati dal base topic e dal proprio nome.
Su HA di default non devi fare niente per il topic di autodiscovery. Se hai già impostato l'integrazione mqtt avrai probabilmente già messo il topic `homeassistant` (che è lo stesso che di default è impostato su zigbee2mqtt). Devi solo abilitare autodiscovery su HA e le due istanze di zigbee2mqtt...facendo attenzione a non dare lo stesso nome ai device su zigbee2mqtt, altrimenti ti incasini.

Comunque trovi tutto sulle doc ufficiali:

https://www.zigbee2mqtt.io/guide/configuration/homeassistant.html

https://www.home-assistant.io/integrations/mqtt/

Non servono guide...serve solo che ti chiarisci i concetti
